component/core/src/main/java/org/exoplatform/social/core/metadata/storage/MetadataStorage.java

```diff
@@ -217,20 +218,17 @@ public List<MetadataItem> getMetadataItemsByMetadataNameAndTypeAndSpaceIds(Strin
     return metadataItemEntities.stream().map(this::fromEntity).toList();
   }
 
-  public List<MetadataItem> getMetadataItemsByMetadataNameAndTypeAndObjectAndSpaceIds(String metadataName,
-                                                                                     String metadataTypeName,
-                                                                                     String objectType,
-                                                                                     List<Long> spaceIds,
-                                                                                     long offset,
-                                                                                      long limit) {
-    MetadataType metadataType = getMetadataTypeWithCheck(metadataTypeName);
-    List<MetadataItemEntity> metadataItemEntities =
-                                                  metadataItemDAO.getMetadataItemsByMetadataNameAndTypeAndObjectAndSpaceIds(metadataName,
-                                                                                                                            metadataType.getId(),
-                                                                                                                            objectType,
-                                                                                                                            spaceIds,
-                                                                                                                            offset,
-                                                                                                                            limit);
+  public List<MetadataItem> getMetadataItemsByFilter(MetadataFilter metadataFilter, long offset, long limit) {
+    MetadataType metadataType = getMetadataTypeWithCheck(metadataFilter.getMetadataTypeName());
+    List<MetadataItemEntity> metadataItemEntities = new ArrayList<>();
+    if (!metadataFilter.getMetadataObjectTypes().isEmpty() && !metadataFilter.getMetadataSpaceIds().isEmpty()) {
+      metadataItemEntities = metadataItemDAO.getMetadataItemsByMetadataNameAndTypeAndObjectsAndSpaceIds(metadataFilter.getMetadataName(),
```

Comment:
can we do a generic query in dao based on the filter ? beucase like that we will not be able to fetch on properties for example